fs/vfs/fs_read.c

```diff
@@ -126,18 +113,21 @@ static ssize_t file_readv_compat(FAR struct file *filep, FAR struct uio *uio)
  *
  * Input Parameters:
  *   filep  - File structure instance
- *   uio    - User buffer information
+ *   iov    - User-provided iovec to save the data
+ *   iovcnt - The number of iovec
  *
  * Returned Value:
  *   The positive non-zero number of bytes read on success, 0 on if an
  *   end-of-file condition, or a negated errno value on any failure.
  *
  ****************************************************************************/
 
-ssize_t file_readv(FAR struct file *filep, FAR struct uio *uio)
+ssize_t file_readv(FAR struct file *filep,
+                   FAR const struct iovec *iov, int iovcnt)
 {
   FAR struct inode *inode;
   ssize_t ret = -EBADF;
+  struct uio uio;
```

Comment:
i think it's better to move this to where it's used. ie. the `if (inode->u.i_ops->readv)` block. what do you think?
ditto for writev.

Reply:
Done